server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalExtendedStats.java

```diff
@@ -250,7 +264,7 @@ public InternalExtendedStats reduce(List<InternalAggregation> aggregations, Redu
         }
         final InternalStats stats = super.reduce(aggregations, reduceContext);
         return new InternalExtendedStats(name, stats.getCount(), stats.getSum(), stats.getMin(), stats.getMax(), sumOfSqrs, sigma,
-            format, getMetadata());
+            m2, format, getMetadata());
```

Comment:
This'll cause the variance to be end up calculated just from the aggregation doing the reducing which seems wrong. Is there a good way to merge m2? In the same way that the sumOfSqrs is combined above with a compensated sum. I *think* https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Parallel_algorithm describes a way to do the merge.

Reply:
I also think this has to handle when one of the nodes doesn't provide `m2`. I *think* the best you can do is fall back to `m2` being `NaN`

Reply:
Fixed this.  
PS:  The actual code added for this suggestion is like 5 lines above this, so it's not showing in the preview mode.

Reply:
Thanks!